sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineOperationsTests.java

```diff
@@ -1998,6 +2002,345 @@ private CreatablesInfo prepareCreatableVirtualMachines(
         return creatablesInfo;
     }
 
+    @Test
+    public void canDeleteSinglePublicIpAddressWithVM() {
+        PublicIpAddress publicIPAddress =
+            networkManager
+                .publicIpAddresses()
+                .define(generateRandomResourceName("pip-", 15))
+                .withRegion(region)
+                .withNewResourceGroup(rgName)
+                .withDynamicIP()
+                .create();
+
+        VirtualMachine vm = computeManager.virtualMachines()
+            .define(vmName)
+            .withRegion(region)
+            .withNewResourceGroup(rgName)
+            .withNewPrimaryNetwork("10.0.0.0/28")
+            .withPrimaryPrivateIPAddressDynamic()
+            .withExistingPrimaryPublicIPAddress(publicIPAddress)
```

Comment:
Should we do `.withExistingPrimaryPublicIPAddress(publicIPAddress, deleteOptions)`?